File: baekjoon/15654.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int N, M;
bool visit[8]{};

int arr[8]{};
vector<int> v;

void dfs(int level) {
    
    if(level == M) {
        for(int i = 0; i < M; i++) cout << arr[i] << " ";
        cout << "\n";
        return;
    }
    
    for(int i = 0; i < N; i++) {
        if(visit[i] == true) continue; 
        visit[i] = true;
        arr[level] = v[i];
        dfs(level + 1);
        visit[i] = false;
    }
}

int main()
{
    cin >> N >> M;
    
    int tmp;
    for(int i = 0; i < N; i++){
        cin >> tmp;
        v.push_back(tmp);
    }
    sort(v.begin(), v.end());
    
    dfs(0);
    return 0;
}

File: baekjoon/15654.py
n, m = map(int, input().split())
input_arr = list(map(int, input().split()))
input_arr.sort()

used = [False] * n
arr = [0] * m

def dfs(level):
    if level == m:
        print(*arr)
        return

    for i in range(n):
        if not used[i] :
            arr[level] = input_arr[i]
            used[i] = True
            dfs(level + 1)
            used[i] = False

dfs(0)